owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt

```diff
@@ -237,3 +244,44 @@ fun Activity.hideSoftKeyboard() {
         )
     }
 }
+
+fun Activity.checkPasscodeEnforced(securityEnforced: ISecurityEnforced) {
+    val sharedPreferencesProvider = SharedPreferencesProviderImpl(this)
+
+    val passcodeEnforced = this.resources.getBoolean(R.bool.passcode_enforced)
+    val passcodeConfigured = sharedPreferencesProvider.getBoolean(PassCodeActivity.PREFERENCE_SET_PASSCODE, false)
+    val patternConfigured = sharedPreferencesProvider.getBoolean(PatternActivity.PREFERENCE_SET_PATTERN, false)
+
+    if (passcodeEnforced && !passcodeConfigured && !patternConfigured) {
+
+        val options = arrayOf(getString(R.string.security_enforced_first_option), getString(R.string.security_enforced_second_option))
+        var optionSelected = 0
+
+        AlertDialog.Builder(this).apply {
+            setCancelable(false)
+            setTitle(getString(R.string.security_enforced_title))
+            setSingleChoiceItems(options, optionSelected) { _, which -> optionSelected = which }
+            setPositiveButton(android.R.string.ok) { dialog, _ ->
+                when (optionSelected) {
+                    0 -> securityEnforced.optionLockSelected(LockType.PASSCODE)
+                    1 -> securityEnforced.optionLockSelected(LockType.PATTERN)
+                }
+                dialog.dismiss()
+            }
+        }.show()
+    }
+}
+
+fun Activity.manageOptionLockSelected(type: LockType) {
+    when (type) {
+        LockType.PASSCODE -> startActivity(Intent(this, PassCodeActivity::class.java).apply {
+            action = PassCodeActivity.ACTION_REQUEST_WITH_RESULT
+            putExtra(EXTRAS_PASSCODE_ENFORCED, true)
+        })
+        LockType.PATTERN -> startActivity(Intent(this, PatternActivity::class.java).apply {
+            action = PatternActivity.ACTION_REQUEST_WITH_RESULT
+            putExtra(EXTRAS_PASSCODE_ENFORCED, true)
```

Comment:
Same as the previous comment. It may sound strange that inside the pattern branch you activate the `EXTRAS_PASSCODE_ENFORCED` flag 😁